test/jdk/java/io/File/TempDirDoesNotExist.java

```diff
@@ -25,91 +25,173 @@
+                                "TempDirDoesNotExist", "nio"
+                            });
+        list.add(args);
+
+        // valid custom java.io.tmpdir
+        args = Arguments.of(0, WARNING,
+                            new String[] {
+                                "-Djava.io.tmpdir=" + USER_DIR,
+                                "TempDirDoesNotExist", "io"
+                            });
+        list.add(args);
+        args = Arguments.of(0, WARNING,
+                            new String[] {
+                                "-Djava.io.tmpdir=" + USER_DIR,
+                                "TempDirDoesNotExist", "nio"
+                            });
+        list.add(args);
+
+        return list.stream();
+    }
+
+    public static Stream<Arguments> counterProvider() {
+        List<Arguments> list = new ArrayList<Arguments>();
+
+        // standard test with default setting for java.io.tmpdir
+        Arguments args = Arguments.of(0,
+                                      new String[] {
+                                          "-Djava.io.tmpdir=" + tempDir(),
+                                          "TempDirDoesNotExist", "io", "nio"
+                                      });
+        list.add(args);
+
+        return list.stream();
     }
 
-    private static void testMessageExist(int exitValue, String errorMsg, String... options) throws Exception {
-        ProcessTools.executeTestJvm(options).shouldContain(errorMsg)
-                .shouldHaveExitValue(exitValue);
+    @ParameterizedTest
+    @MethodSource("existingProvider")
+    public void existingMessage(int exitValue, String errorMsg,
+                                String... options) throws Exception {
+       ProcessTools.executeTestJvm(options).shouldContain(errorMsg)
+           .shouldHaveExitValue(exitValue);
     }
 
-    private static void testMessageNotExist(int exitValue, String errorMsg,String... options) throws Exception {
-        ProcessTools.executeTestJvm(options).shouldNotContain(errorMsg).shouldHaveExitValue(exitValue);
+    @ParameterizedTest
+    @MethodSource("nonexistentProvider")
+    public void nonexistentMessage(int exitValue, String errorMsg,
+                                   String... options) throws Exception {
+        ProcessTools.executeTestJvm(options).shouldNotContain(errorMsg)
+            .shouldHaveExitValue(exitValue);
     }
 
-    private static void testMessageCounter(int exitValue,String... options) throws Exception {
+    @ParameterizedTest
+    @MethodSource("counterProvider")
+    public  void messageCounter(int exitValue, String... options)
+        throws Exception {
         OutputAnalyzer originalOutput = ProcessTools.executeTestJvm(options);
         List<String> list = originalOutput.asLines().stream().filter(line
-                -> line.equalsIgnoreCase(ioWarningMsg)).collect(Collectors.toList());
+                -> line.equalsIgnoreCase(WARNING)).collect(Collectors.toList());
```

Comment:
`Stream.toList()` can bu applied here

Reply:
So changed in 50411b85aeefb2576a149ae126b7d55aad6e2be0.